src/elements/e-video.vue

```diff
@@ -97,108 +85,126 @@
 
       /**
        * Allows to set the video player width.
+       * width can either be responsive or hard coded.
+       * youtube standard -> width="560".
        */
       width: {
-        type: [String, Number],
-        default: 560,
+        type: String,
+        default: null,
       },
 
       /**
        * Allows to set the video player height.
+       * height can either be responsive or hard coded.
+       * youtube standard -> height="315".
        */
       height: {
-        type: [String, Number],
-        default: 315,
-      },
-
-      /**
-       * Defines the players size to be responsive 100% width to the parent container
-       */
-      responsive: {
-        type: Boolean,
-        default: false,
+        type: String,
+        default: null,
       },
     },
 
     // emits: {},
     // setup(): Setup {},
 
     computed: {
+      /**
+       * Returns a validity check for the video url.
+       */
+      validateUrl(): boolean {
+        if (this.videoUrl) {
+          /**
+           * This is for sources with multiple allowed urls.
+           * e.g. youtube (standard, short)
+           */
+          if (Array.isArray(VideoUrlSource[this.source])) {
+            const videoSources = VideoUrlSource[this.source] as string[];
+
+            return this.arrayHasString(videoSources, this.videoUrl);
+          }
+
+          if (!this.videoUrl.startsWith(VideoUrlSource[this.source] as string)) {
+            // eslint-disable-next-line no-console
+            console.error('Invalid video url', this.$el);
+
+            return false;
+          }
+
+          return true;
+        }
+
+        return true;
+      },
+
+      /**
+       * Returns the validated and adapted video source.
+       */
       mapSrc(): string | undefined {
-        const playerBaseUrl = playerUrlSources[this.source];
+        const playerBaseUrl = PlayerUrlSource[this.source];
 
-        if (!(this.videoId || this.videoUrl)) {
+        if (!this.videoId && !this.videoUrl) {
           // eslint-disable-next-line no-console
           console.error("Neither 'videoId' or 'videoUrl' were defined", this.$el);
 
           return undefined;
         }
 
         if (this.videoUrl && !this.videoId) {
-          if (!this.validateUrl()) {
+          if (!this.validateUrl) {
             return undefined;
           }
 
-          if (this.source === 'youtube') {
-            /**
-             * Extract id from video url
-             */
-            const id = this.videoUrl.substring(
-              this.videoUrl.indexOf('v=') + 2,
-              this.videoUrl.indexOf('&')
-            );
+          let id: string | null;
```

Comment:
suggestion: As you did suggest to me, lets go with `undefined` instead of `null`. I think this will also allow to just use `let id: string;`, since it is already undefined.

Reply:
I did go with the undefined approach, had to use the [Nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) because `new URLSearchParams().get()` does return null if nothing is found.